frontend/src/views/Options.vue

```diff
@@ -297,14 +327,48 @@ export default Vue.extend({
     },
 
     async setCurrentChain() {
+      console.log(+getConfigValue('VUE_APP_NETWORK_ID'));
       localStorage.setItem('currentChain', this.currentChain);
       this.updateCurrentChainSupportsMerchandise();
       this.updateCurrentChainSupportsPvP();
       this.updateCurrentChainSupportsQuests();
       Events.$emit('setting:currentChain', { value: this.currentChain });
       addChainToRouter(this.currentChain);
-      await this.configureMetaMask(+getConfigValue('VUE_APP_NETWORK_ID'));
+      if(!this.connectingWalletConnect) await this.configureMetaMask(+getConfigValue('VUE_APP_NETWORK_ID'));
     },
+    async connectWallet() {
+      if(this.isWalletConnect) {
+        localStorage.removeItem('walletconnect');
+        window.location.reload();
+      }
+      else{
+        this.connectingWalletConnect = true;
+        this.currentChain = this.walletConnectChain;
+        this.setCurrentChain();
+
+        console.log('connecting to net ',+getConfigValue('VUE_APP_NETWORK_ID'));
+
+        const rpcs = {} as RPCS;
+        config.supportedChains.forEach((chain) => {
+          const chainId = getConfigValue('VUE_APP_NETWORK_ID', chain);
+          rpcs[chainId] = getConfigValue('rpcUrls',chain)[0];
+        });
+
+        //  Create WalletConnect Provider
+        const provider = new WalletConnectProvider({
+          rpc: rpcs,
+          chainId: +getConfigValue('VUE_APP_NETWORK_ID'),
+        });
+
+        //  Enable session (triggers QR Code modal)
+        await provider.enable();
+
+        const w3 = new Web3(provider as any);
+        this.setWeb3(w3);
+        (this as any).$router.push({ path: '/', query: { chain: this.currentChain } });
+        window.location.reload();
```

Comment:
Is a force reload necessary?

Reply:
Yes

Reply:
If we don't reload here, MM would be trying to reconnect. Contracts and the whole state needs to be reloaded anyways, because the network might change